### PR TITLE
Added tests for rotation operations

### PR DESCRIPTION
**Pull request by runfalk**

### CODE REVIEW

File: src/imageops/transform.rs [hunk 1]
```diff
@@ -49,16 +49,175 @@ pub fn transpose(buf: &OpBuffer) -> OpBuffer {
...
+  match orientation.to_flips() {
     (false, false, false) => buf.clone(),
     (false, false, true) => flip_vertical(buf),
     (false, true, false) => flip_horizontal(buf),
     (false, true, true) => flip_horizontal(&flip_vertical(buf)),
     (true, false, false) => transpose(buf),
     (true, false, true) => flip_vertical(&transpose(buf)),
     (true, true, false) => flip_horizontal(&transpose(buf)),
-    (true, true, true) => flip_vertical(&transpose(&flip_horizontal(buf))),
+    (true, true, true) => flip_vertical(&flip_horizontal(&transpose(buf))),
```
Comment (pedrocr) [inline src/imageops/transform.rs]: Why was this needed? Is transpose always the first operation?

File: src/imageops/transform.rs [hunk 2]
```diff
@@ -49,16 +49,175 @@ pub fn transpose(buf: &OpBuffer) -> OpBuffer {
...
+  rotate_buffer(buf, &img.orientation)
+}
+
+#[cfg(test)]
+mod tests {
+  use decoders::Orientation;
+  use imageops::OpBuffer;
+  use super::rotate_buffer;
+
+  /// Helper function to allow human readable creation of `OpBuffer` instances
```
Comment (pedrocr) [inline src/imageops/transform.rs]: This should probably be in mod.rs instead so anything can use it

File: src/imageops/transform.rs (hunk 1 above)
Comment (runfalk) [inline src/imageops/transform.rs]: Apparently order matters as far as transposing is concerned. The previous code was wrong.

Comment (pedrocr) [inline src/imageops/transform.rs]: That's good news as it means the new code should be easier to write as transpose is always the first op if it exists.
